df.ui-menus.xml

```diff
@@ -2663,30 +2663,60 @@
         <int32_t name='demand_period'/>
     </struct-type>
 
+    <struct-type type-name='markup_text_wordst'>
+        read and write serialization is in order as below
+        <stl-string name='str'/>
+        <uint8_t name='red' init-value='255'/>
+        <uint8_t name='green' init-value='255'/>
+        <uint8_t name='blue' init-value='255'/>
+        <int32_t name='link_index' init-value='-1'/>
+        <int32_t name='px' init-value='-1'/>
+        <int32_t name='py' init-value='-1'/>
```

Comment:
**Comments:**
link_index: refers to this text box's link vector, makes clickable link in legends mode
px: offset from left of text box in characters
py: offset from top of text box (positive is downwards)